src/meson.build

```diff
@@ -264,146 +264,46 @@ if not(get_option('build_static'))
 	endif
 endif
 
-libbrotli = []
-if get_option('with_brotli')
-	libbrotli = [ compiler.find_library('brotlienc') ]
-	if compiler.has_function('BrotliEncoderCreateInstance', args: defs, dependencies: libbrotli, prefix: '#include <brotli/encode.h>')
-		conf_data.set('HAVE_BROTLI_ENCODE_H', true)
-		conf_data.set('HAVE_BROTLI', true)
-	else
-		error('Couldn\'t find brotli header / library')
-	endif
-endif
+libbrotli = dependency('libbrotlienc', required: get_option('with_brotli'))
+conf_data.set('HAVE_BROTLI_ENCODE_H', libbrotli.found())
+conf_data.set('HAVE_BROTLI', libbrotli.found())
 
-libbz2 = []
-if get_option('with_bzip')
-	libbz2 = [ compiler.find_library('bz2') ]
-	if compiler.has_function('BZ2_bzCompress', args: defs, dependencies: libbz2, prefix: '#include <bzlib.h>')
-		conf_data.set('HAVE_BZLIB_H', true)
-		conf_data.set('HAVE_LIBBZ2', true)
-	else
-		error('Couldn\'t find bz2 header / library')
-	endif
-endif
+libbz2 = dependency('bzip2', required: get_option('with_bzip'))
```

Comment:
Unfortunately, this doesn't reliably work as upstream doesn't ship one. :(

The previous find_library() check is correct, here. But you can pass a feature option to find_library(..., required: ...) too.

https://sourceware.org/pipermail/bzip2-devel/2022q2/000169.html

Reply:
Hrm fedora does